### DISCUSSION TITLE
How to connect Hygraph properly?

### DISCUSSION THREAD

**Discussion by NICCSAMTEN** [webstudio-is/webstudio-community#182] [Q&A]
So, I want to connect to Hygraph. But its always shown like this

![image](https://github.com/user-attachments/assets/bb79f38d-5e0c-45b2-adb3-4394415d0241)

Does anybody know how to fix this? I'm just following the tutorial but I got stuck here.

Also, here is in the Hygraph and its working because its fetching the data.

![image](https://github.com/user-attachments/assets/ce51955b-0860-4255-9708-6971b70f3ed8)

**Comment by NICCSAMTEN** [comment]
And now there isn't any data displayed here

![image](https://github.com/user-attachments/assets/467dc329-3b9f-4afb-9424-07d82ba9fecf)

**Comment by johnsicili** [comment]
The problem is in the query. The query needs to accept the slug variable. In the query builder, instead of typing slug, there should be a `$` you select.

<img width="723" alt="Screenshot 2024-09-06 at 7 44 43 AM" src="https://github.com/user-attachments/assets/2cf188d6-b87e-4d80-8e35-097a1882508c">

When I clicked the dollar sign, it put `query MyQuery($slug: String = "") {` in my query which lets it accept `$slug`.

Hope that helps. Start there and see if it works. Let me know if you'd like any clarification.

**Comment by NICCSAMTEN** [comment]
Why i cannot get my title?
![image](https://github.com/user-attachments/assets/385c1220-2369-48af-a7c0-22d0f6d82811)
![image](https://github.com/user-attachments/assets/11dfe2ca-da0f-4538-93cb-31d81194c87b)

**Reply by kof** [reply]
you have provided the post object, not the title in it `posts[0].title` would be it 
also posts is an array and you should use the collection for iterating over it https://docs.webstudio.is/university/core-components/collection.md

**Comment by NICCSAMTEN** [comment]
do you know how to remove the HTML elements?
![image](https://github.com/user-attachments/assets/0e38b034-fbac-496f-a31d-9f8bee6b6e70)

![image](https://github.com/user-attachments/assets/1fb8101a-35d0-413f-953b-cf48af399a65)

**Reply by kof** [reply]
how did you get it in there? data bound as text content on a box? if yes you need to Content Embed instead
